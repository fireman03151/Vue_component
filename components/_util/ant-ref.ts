import { nextTick } from 'vue';
const antvRef = {
  beforeMount: function bind(el, binding, vnode) {
    nextTick(function() {
      binding.value(el, vnode.key);
    });
    binding.value(el, vnode.key);
  },
  updated: function update(el, binding, vnode, oldVnode) {
    if (oldVnode && oldVnode.directives) {
      let oldBinding = oldVnode.directives.find(function(directive) {
        return directive === antvRef;
      });
      if (oldBinding && oldBinding.value !== binding.value) {
        oldBinding && oldBinding.value(null, oldVnode.key);
        binding.value(el, vnode.key);
        return;
      }
    }
    // Should not have this situation
    if (vnode.el !== oldVnode.el) {
      binding.value(el, vnode.key);
    }
  },
  unmounted: function unbind(el, binding, vnode) {
    binding.value(null, vnode.key);
  },
};

export default antvRef;
