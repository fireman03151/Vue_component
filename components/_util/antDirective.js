import ref from 'vue-ref';
// import { antInput } from './antInputDirective';
import { antDecorator } from './FormDecoratorDirective';
import { antPortal } from './portalDirective';

export default {
  install: Vue => {
    Vue.use(ref, { name: 'ant-ref' });
    antDecorator(Vue);
    antPortal(Vue);
  },
};
