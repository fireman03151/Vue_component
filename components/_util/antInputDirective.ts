function onCompositionStart(e: Event) {
  (e.target as any).composing = true;
}

function onCompositionEnd(e: Event) {
  const target = e.target as any;
  if (target.composing) {
    target.composing = false;
    trigger(target, 'input');
  }
}

function trigger(el: HTMLElement, type: string) {
  const e = document.createEvent('HTMLEvents');
  e.initEvent(type, true, true);
  el.dispatchEvent(e);
}

export function addEventListener(
  el: Element,
  event: string,
  handler: EventListener,
  options?: EventListenerOptions,
) {
  el.addEventListener(event, handler, options);
}
const antInput = {
  created(el: Element, binding: { modifiers: { lazy: any } }) {
    if (!binding.modifiers || !binding.modifiers.lazy) {
      addEventListener(el, 'compositionstart', onCompositionStart);
      addEventListener(el, 'compositionend', onCompositionEnd);
      // Safari < 10.2 & UIWebView doesn't fire compositionend when
      // switching focus before confirming composition choice
      // this also fixes the issue where some browsers e.g. iOS Chrome
      // fires "change" instead of "input" on autocomplete.
      addEventListener(el, 'change', onCompositionEnd);
    }
  },
};

export default antInput;
