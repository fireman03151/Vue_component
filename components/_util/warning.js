import warning from 'warning'

const warned = {}
export default (valid, message) => {
  if (process.env.NODE_ENV !== 'production') {
    if (!valid && !warned[message]) {
      warning(false, message)
      warned[message] = true
    }
  }
}
