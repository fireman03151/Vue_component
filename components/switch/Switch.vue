<template>
  <span>
    <span class={`${prefixCls}-inner`}>
      <component></component>
    </span>
  </span>
</template>
<script>
export default {
  name: 'Radio',
  props: {
    prefixCls: {
      default: 'ant-radio',
      type: String,
    },
    defaultChecked: Boolean,
    checked: { type: Boolean, default: undefined },
    disabled: Boolean,
    isGroup: Boolean,
    value: [String, Number, Boolean],
    name: String,
    onGroupChange: Function,
  },
  model: {
    prop: 'checked',
  },
  data () {
    const { checked, defaultChecked } = this
    return {
      stateChecked: checked === undefined ? defaultChecked : checked,
    }
  },
  computed: {
    hasDefaultSlot () {
      return !!this.$slots.default
    },
    classes () {
      const { prefixCls, disabled, stateChecked } = this
      return {
        [`${prefixCls}-wrapper`]: true,
        [`${prefixCls}-wrapper-checked`]: stateChecked,
        [`${prefixCls}-wrapper-disabled`]: disabled,
      }
    },
    checkboxClass () {
      const { prefixCls, disabled, stateChecked } = this
      return {
        [`${prefixCls}`]: true,
        [`${prefixCls}-checked`]: stateChecked,
        [`${prefixCls}-disabled`]: disabled,
      }
    },
  },
  methods: {
    handleChange (event) {
      const targetChecked = event.target.checked
      const { name, value, checked } = this
      if (checked === undefined) {
        this.stateChecked = targetChecked
      }
      this.$emit('input', targetChecked)
      const target = {
        name,
        value,
        checked: targetChecked,
      }
      this.$emit('change', {
        target,
        stopPropagation () {
          event.stopPropagation()
        },
        preventDefault () {
          event.preventDefault()
        },
      })
      if (this.isGroup) {
        this.onGroupChange({ target })
      }
    },
  },
  watch: {
    checked (val) {
      this.stateChecked = val
    },
  },
}
</script>
