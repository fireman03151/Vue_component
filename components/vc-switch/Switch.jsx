import { switchPropTypes } from './PropTypes';
import BaseMixin from '../_util/BaseMixin';
import { hasProp, getOptionProps, getComponentFromProp, getListeners } from '../_util/props-util';

// function noop () {
// }
export default {
  name: 'VcSwitch',
  mixins: [BaseMixin],
  model: {
    prop: 'checked',
    event: 'change',
  },
  props: {
    ...switchPropTypes,
    prefixCls: switchPropTypes.prefixCls.def('rc-switch'),
    // onChange: switchPropTypes.onChange.def(noop),
    // onClick: switchPropTypes.onClick.def(noop),
  },
  data() {
    let checked = false;
    if (hasProp(this, 'checked')) {
      checked = !!this.checked;
    } else {
      checked = !!this.defaultChecked;
    }
    return {
      stateChecked: checked,
    };
  },
  watch: {
    checked(val) {
      this.stateChecked = val;
    },
  },
  mounted() {
    this.$nextTick(() => {
      const { autoFocus, disabled } = this;
      if (autoFocus && !disabled) {
        this.focus();
      }
    });
  },
  methods: {
    setChecked(checked, e) {
      if (this.disabled) {
        return;
      }
      if (!hasProp(this, 'checked')) {
        this.stateChecked = checked;
      }
      this.$emit('change', checked, e);
    },
    handleClick(e) {
      const checked = !this.stateChecked;
      this.setChecked(checked, e);
      this.$emit('click', checked, e);
    },
    handleKeyDown(e) {
      if (e.keyCode === 37) {
        // Left
        this.setChecked(false, e);
      } else if (e.keyCode === 39) {
        // Right
        this.setChecked(true, e);
      }
    },
    handleMouseUp(e) {
      if (this.$refs.refSwitchNode) {
        this.$refs.refSwitchNode.blur();
      }
      this.$emit('mouseup', e);
    },
    focus() {
      this.$refs.refSwitchNode.focus();
    },
    blur() {
      this.$refs.refSwitchNode.blur();
    },
  },
  render() {
    const { prefixCls, disabled, loadingIcon, tabIndex, ...restProps } = getOptionProps(this);
    const checked = this.stateChecked;
    const switchClassName = {
      [prefixCls]: true,
      [`${prefixCls}-checked`]: checked,
      [`${prefixCls}-disabled`]: disabled,
    };
    const spanProps = {
      props: { ...restProps },
      on: {
        ...getListeners(this),
        keydown: this.handleKeyDown,
        click: this.handleClick,
        mouseup: this.handleMouseUp,
      },
      attrs: {
        type: 'button',
        role: 'switch',
        'aria-checked': checked,
        disabled,
        tabIndex,
      },
      class: switchClassName,
      ref: 'refSwitchNode',
    };
    return (
      <button {...spanProps}>
        {loadingIcon}
        <span class={`${prefixCls}-inner`}>
          {checked
            ? getComponentFromProp(this, 'checkedChildren')
            : getComponentFromProp(this, 'unCheckedChildren')}
        </span>
      </button>
    );
  },
};
