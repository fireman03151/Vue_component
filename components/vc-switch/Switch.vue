<script>
import { switchPropTypes } from './PropTypes'
import BaseMixin from '../_util/BaseMixin'
import { hasProp, filterEmpty, getOptionProps } from '../_util/props-util'

// function noop () {
// }
export default {
  name: 'vc-switch',
  mixins: [BaseMixin],
  model: {
    prop: 'checked',
    event: 'change',
  },
  props: {
    ...switchPropTypes,
    prefixCls: switchPropTypes.prefixCls.def('rc-switch'),
    checkedChildren: switchPropTypes.checkedChildren.def(null),
    unCheckedChildren: switchPropTypes.unCheckedChildren.def(null),
    defaultChecked: switchPropTypes.defaultChecked.def(''),
    // onChange: switchPropTypes.onChange.def(noop),
    // onClick: switchPropTypes.onClick.def(noop),
  },
  data () {
    let checked = false
    if (hasProp(this, 'checked')) {
      checked = !!this.checked
    } else {
      checked = !!this.defaultChecked
    }
    return {
      stateChecked: checked,
    }
  },
  monted () {
    this.$nextTick(() => {
      const { autoFocus, disabled } = this
      if (autoFocus && !disabled) {
        this.focus()
      }
      this.refSwitchNode = this.$refs.refSwitchNode
    })
  },
  watch: {
    checked (val) {
      this.stateChecked = val
    },
  },
  methods: {
    setChecked (checked) {
      if (this.disabled) {
        return
      }
      if (!hasProp(this, 'checked')) {
        this.stateChecked = checked
      }
      this.$emit('change', checked)
    },
    toggle () {
      const checked = !this.stateChecked
      this.setChecked(checked)
      this.$emit('click', checked)
    },
    handleKeyDown (e) {
      if (e.keyCode === 37) { // Left
        this.setChecked(false)
      } else if (e.keyCode === 39) { // Right
        this.setChecked(true)
      } else if (e.keyCode === 32 || e.keyCode === 13) { // Space, Enter
        this.toggle()
      }
    },
    handleMouseUp (e) {
      if (this.refSwitchNode) {
        this.refSwitchNode.blur()
      }
      this.$emit('mouseUp', e)
    },
    focus () {
      this.refSwitchNode.focus()
    },
    blur () {
      this.refSwitchNode.blur()
    },
    getChildren (slotName) {
      if (hasProp(this, slotName)) {
        return this[slotName]
      }
      if (this.$slots && this.$slots[slotName]) {
        return filterEmpty(this.$slots[slotName])
      }
      return null
    },
  },
  render () {
    const { prefixCls, disabled, tabIndex, ...restProps } = getOptionProps(this)
    const checked = this.stateChecked
    const switchTabIndex = disabled ? -1 : (tabIndex || 0)
    const switchClassName = {
      [prefixCls]: true,
      [`${prefixCls}-checked`]: checked,
      [`${prefixCls}-disabled`]: disabled,
    }
    const childrenType = checked ? 'checkedChildren' : 'unCheckedChildren'
    return (
      <span
        {...restProps}
        class={switchClassName}
        tabIndex={switchTabIndex}
        ref='refSwitchNode'
        onKeydown={this.handleKeyDown}
        onClick={this.toggle}
        onMouseup={this.handleMouseUp}
      >
        <span class={`${prefixCls}-inner`}>
          {this.getChildren(childrenType)}
        </span>
      </span>
    )
  },
}
</script>
