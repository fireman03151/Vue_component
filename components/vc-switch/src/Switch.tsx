import { switchPropTypes } from './PropTypes';
import Omit from 'omit.js';
import { defineComponent, nextTick, onMounted, ref } from 'vue';
import KeyCode from '../../_util/KeyCode';
import { getPropsSlot } from '../../_util/props-util';

export default defineComponent({
  name: 'VcSwitch',
  inheritAttrs: false,
  props: {
    ...switchPropTypes,
    prefixCls: switchPropTypes.prefixCls.def('rc-switch'),
  },
  emits: ['update:checked', 'mouseup', 'change', 'click'],
  setup(props, { attrs, slots, emit, expose }) {
    const checked = ref('checked' in props ? !!props.checked : !!props.defaultChecked);

    const refSwitchNode = ref();

    onMounted(() => {
      nextTick(() => {
        if (props.autofocus && !props.disabled) {
          refSwitchNode.value.focus();
        }
      });
    });

    const setChecked = (check: boolean, e: MouseEvent | KeyboardEvent) => {
      if (props.disabled) {
        return;
      }
      checked.value = !checked.value;
      emit('update:checked', checked);
      emit('change', check, e);
    };

    const handleClick = (e: MouseEvent) => {
      setChecked(checked.value, e);
      emit('click', checked.value, e);
    };

    const handleKeyDown = (e: KeyboardEvent) => {
      if (e.keyCode === KeyCode.LEFT) {
        setChecked(false, e);
      } else if (e.keyCode === KeyCode.RIGHT) {
        setChecked(true, e);
      }
    };

    const handleMouseUp = (e: MouseEvent) => {
      refSwitchNode.value?.blur();
      emit('mouseup', e);
    };

    const focus = () => {
      refSwitchNode.value?.focus();
    };
    const blur = () => {
      refSwitchNode.value?.blur();
    };

    expose({ focus, blur });

    return () => (
      <button
        {...Omit(props, [
          'prefixCls',
          'checkedChildren',
          'unCheckedChildren',
          'checked',
          'autofocus',
          'defaultChecked',
        ])}
        {...attrs}
        onKeydown={handleKeyDown}
        onClick={handleClick}
        onMouseup={handleMouseUp}
        type="button"
        role="switch"
        aria-checked={checked.value}
        disabled={props.disabled}
        class={{
          [attrs.class as string]: attrs.class,
          [props.prefixCls]: true,
          [`${props.prefixCls}-checked`]: checked.value,
          [`${props.prefixCls}-disabled`]: props.disabled,
        }}
        ref={refSwitchNode}
      >
        {props.loadingIcon}
        <span class={`${props.prefixCls}-inner`}>
          {checked.value
            ? getPropsSlot(slots, props, 'checkedChildren')
            : getPropsSlot(slots, props, 'unCheckedChildren')}
        </span>
      </button>
    );
  },
});
