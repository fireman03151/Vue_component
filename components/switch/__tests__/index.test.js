import Switch from '..';
import { mount } from '@vue/test-utils';
import focusTest from '../../../tests/shared/focusTest';
import { resetWarned } from '../../_util/warning';
import mountTest from '../../../tests/shared/mountTest';

describe('Switch', () => {
  focusTest(Switch);
  mountTest(Switch);

  it('should has click wave effect', async () => {
    const wrapper = mount(Switch);
    wrapper.find('.ant-switch').trigger('click');
    await new Promise(resolve => setTimeout(resolve, 0));
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('warning if set `value`', () => {
    resetWarned();

    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
    mount(Switch, { props: { value: '' } });
    expect(errorSpy).toHaveBeenCalledWith(
      'Warning: [antdv: Switch] `value` is not validate prop, do you mean `checked`?',
    );
    errorSpy.mockRestore();
  });
});
