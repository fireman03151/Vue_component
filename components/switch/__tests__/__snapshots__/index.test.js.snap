// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should has click wave effect 1`] = `
<button class="ant-switch ant-switch-checked" type="button" role="switch" aria-checked="true">
  <!----><span class="ant-switch-inner"><!----></span>
</button>
`;
